src/main/java/tool/rental/domain/use_cases/UpdateToolUseCase.java

```diff
@@ -0,0 +1,32 @@
+package tool.rental.domain.use_cases;
+
+import tool.rental.domain.entities.Tool;
+import tool.rental.domain.repositories.ToolRepository;
+import tool.rental.utils.ToastError;
+
+public class UpdateToolUseCase {
+    private final ToolRepository toolRepository = new ToolRepository();
+
+    public void execute (Tool tool, String brand, String name, double cost) throws ToastError {
+        if (brand == null || brand.isEmpty()) {
+            throw new ToastError("Marca da ferramenta não pode ser nulo.", "Campo não pode ser nulo"
+            );
+        }
+        if (name == null || name.isEmpty()) {
+            throw new ToastError("Nome da ferramenta não pode ser nulo.", "Campo não pode ser nulo"
+            );
+        }
+        if (cost == 0.00) {
```

Comment:
```suggestion
        if (cost <= 0.00) {
```

Assim evita de o usuário informar um valor negativo também